tenzoMath/fanucModel.cpp

```diff
@@ -0,0 +1,326 @@
+#include "fanucModel.h"
+#include <iostream>
+#include <opencv2/core.hpp>
+#include <vector>
+#include "poly34.h"
+#define PI 3.14159265
+
+FanucModel::FanucModel()
+    :
+    RoboModel(std::vector<std::array<double, 4>>{
+        {0, 0, 150, PI / 2},
+        {0, 0, 790, 0},
+        {0, 0, 250, PI / 2},
+        {835, 0, 0, -PI / 2},
+        {0, 0, 0, PI / 2},
+        {100, 0, 0, 0},
+        {130, PI / 2, -90, 0},
+        {-190, 0, 0, 0}
+    }),
+    _toCamera((cv::Mat_<double>(4, 4) << 0, -1, 0, -43, 1, 0, 0, -90, 0, 0, 1, 130, 0, 0, 0, 1)),
+    _toSixth((cv::Mat_<double>(4, 4) << 1, 0, 0, 0, 0, 1, 0, 0, 0, 0, 1, -190, 0, 0, 0, 1)),
+    _forMovingToCamera((cv::Mat_<double>(4, 4) << 0, -1, 0, 43, 1, 0, 0, 90, 0, 0, 1, 130, 0, 0, 0, 1))
+{
+}
+
+std::vector<RoboModel::DhParameters> FanucModel::getDhParameters() const
+{
+    return _kinematicChain;
+}
+
+std::vector<double> FanucModel::jointsToQ(std::array<double, 6> j)
+{
+    //degrees to radians
+    for (int i = 0; i < 6; ++i)
+    {
+        j[i] *= PI / 180.0;
+    }
+    std::vector<double> q;
+    q.resize(6);
+    q[0] = j[0];
+    q[1] = -j[1] + PI / 2;
+    q[2] = j[2] + j[1];
+    q[3] = -j[3];
+    q[4] = j[4];
+    q[5] = -j[5];
```

Comment:
(**must be changed**): Inefficient way to create a `std::vector`: 
1) you allocate memory for 6 `double` **and** initialize them (because of `resize()` and they all equal `0.0`);
2) you reassing all values of container.

Possible desicion this problem:
```C++
std::vector<double> result;
result.reserve(6);
result.push_back(j.at(0)); // I recommend to use method `at()` instead of `operator[]()` because it more safety.
/* etc */
```

Reply:
Also I recommend to change names of your variables. It's too difficult to read such code.